assets/js/components/ContentAside.js

```diff
@@ -46,105 +49,16 @@ module.exports = class ContentAside {
-    if (!this.$contentHeader) {
-      return;
-    }
-
-    this.asidePaddingRight = 0;
-
-    if (this.isViewportWideWithContentAside()) {
-      this.asidePaddingRight = 4;
-
-      // Detect aside scollbar width
-      this.$elm.classList.add(this.cssStickyClassName);
-      this.scrollbarWidth = this.$elm.offsetWidth - this.$elm.clientWidth;
-      this.$elm.classList.remove(this.cssStickyClassName);
-
-      this.$elm.style.marginRight = (this.scrollbarWidth * -1) + 'px';
-      this.$elm.style.paddingRight = this.scrollbarWidth + this.asidePaddingRight + 'px';
-    }
-
-    let topOfContentHeader = this.$contentHeader.getBoundingClientRect().top;
-
-    // If it's position is sticky
-    if (this.$elm.classList.contains(this.cssStickyClassName)) {
-
-      // If Contextual Data shows on the screen then remove sticky aside
-      if (topOfContentHeader > 0) {
-        this.$elm.classList.remove(this.cssStickyClassName);
-        this.$elm.style.paddingRight = this.scrollbarWidth + this.asidePaddingRight + 'px';
-        return;
-      }
-
-      // Allow it to scroll again if it would otherwise over-/under-lay following element
-      let bottomOfMain = this.mainTarget.getBoundingClientRect().bottom;
-      if (bottomOfMain < this.$elm.offsetHeight) {
-        let amountToNudgeUp = bottomOfMain - this.$elm.offsetHeight;
-        this.$elm.style.top = amountToNudgeUp + 'px';
-        return;
-      }
-
-      // Ensure top of component is not off top of screen once bottom of main is off screen bottom
-      // Safety net: required because a fast scroll may prevent all code running as desired.
-      if (bottomOfMain >= this.window.innerHeight) {
-        this.$elm.style.top = '0px';
-      }
-
-      return;
-    }
-
-    // Otherwise stick its position if it would otherwise scroll off the top of the screen
-    if (topOfContentHeader < 0) {
-      this.$elm.classList.add(this.cssStickyClassName);
-      this.$elm.style.top = '0px';
-      this.$elm.style.paddingRight = this.asidePaddingRight + 'px';
-    }
-  }
-
-  isViewportWideWithContentAside() {
-    return this.window.matchMedia('(min-width: 900px)').matches;
+  handleSpaceWithScrollBar() {
+    let scrollbarWidth = this.$elm.offsetWidth - this.$elm.clientWidth;
+    this.addAdditionalMarginIfScrollBarIsVisible(scrollbarWidth);
   }
 
-  handleResize(scrollingHandler, scrollingHandlerImmediate) {
-    const isViewportWideWithContentAside = this.isViewportWideWithContentAside();
-    if (!this.isScrollingHandled && isViewportWideWithContentAside) {
-      this.startHandlingScrolling(scrollingHandler, scrollingHandlerImmediate);
-    } else if (this.isScrollingHandled && !isViewportWideWithContentAside) {
-      this.stopHandlingScrolling(scrollingHandler);
+  addAdditionalMarginIfScrollBarIsVisible(scrollbarWidth) {
+    if (scrollbarWidth > 0) {
+      this.$elm.style.marginRight = (scrollbarWidth * -1) + 'px';
+    } else {
+      this.$elm.style.marginRight = '0px';
```

Comment:
Is it not possible to dynamically set margins in the CSS? Doing it in JS seems pretty odd to me (Keep in mind I'm very clueless with older school practices!)

Reply:
We don't know the scrollbar width in the CSS (it is different for some browsers).